### PR TITLE
Link and expand time types docs

### LINKED ISSUE #176: Improve documentation for time types
Improve documentation for time types, and link to it from functions that accept time types.

### PR DESCRIPTION
**Pull request by briancavalier**
Close #176

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -11,19 +11,48 @@ API
 Time
 ^^^^
 
+All time-related types use units defined by a :ref:`Clock`. The default :ref:`Scheduler` :ref:`Clock` uses milliseconds as its units: :ref:`Time`, :ref:`Delay <Delay-type>`, :ref:`Period`, and :ref:`Offset` will all be millisecond values.
+
```
Comment (davidchase) [inline docs/api.rst]: if milliseconds are defined here do we need to repeat them for each one individually ?

Comment (briancavalier) [reply docs/api.rst]: There are probably better ways to structure things.  My thinking was that repeating it in multiple spots will be helpful.  I was considering two main ways people use documentation:

1. They read or skim parts of it like a book to understand concepts, or get a feel for the API, etc.
2. They look for the answer answer a specific question, like "what are the units of the number I pass to `delay`?"

Repetition is slightly annoying for number 1, but imho, very helpful for number 2.

For number 1, they'll likely encounter the `Time` heading first, and read that time units are milliseconds when using the default `Scheduler`.  My thinking is that they'll then just tend to ignore the repetition when they encounter it later.

For number 2, someone will tend just to go look for the `delay` function (possibly finding it by readthedocs search, or just find-on-page text search).  At that point they'll find a link to the `Delay` type, hopefully click on it, and see that its units are milliseconds.  If `Delay` doesn't say that it's milliseconds, the person would have to take another step: find the `Time` heading and read the intro prose.

File: docs/api.rst [hunk 2]
```diff
@@ -266,7 +295,7 @@ periodic
 
   periodic :: Period -> Stream void
 
-Create an infinite :ref:`Stream` containing events that occur at a specified period. ::
+Create an infinite :ref:`Stream` containing events that occur at a specified :ref:`Period`. ::
```
Comment (Frikki) [inline docs/api.rst]: Should we add that the first event occurs at time 0?

Comment (briancavalier) [reply docs/api.rst]: In fact, it's specified twice just below: once in the ascii diagram, and once in the sentence below the ascii diagram.  Maybe the right thing is simply to move that lower sentence up to this first paragraph?  E.g.:

> Create an infinite :ref:`Stream` containing events that occur at a specified :ref:`Period`. The first event occurs at time 0, and the event values are undefined. ::

Comment (Frikki) [reply docs/api.rst]: Good idea.

Comment (briancavalier) [reply docs/api.rst]: Done in 3e90ef4

**Pull request merged**